packages/superset-ui-connection/test/callApi/callApi.test.ts

```diff
@@ -8,6 +8,10 @@ import throwIfCalled from '../utils/throwIfCalled';
 import { CallApi } from '../../src/types';
 import { DEFAULT_FETCH_RETRY_OPTIONS } from '../../src/constants';
 
+function getObjectKeys<T>(object: T) {
+  return Object.keys(object) as (keyof T)[];
+}
```

Comment:
If this function is just for typing, maybe it makes sense to just inline it:

```ts
        (Object.keys(object) as (keyof T)[]).forEach(key => {
          expect(body.get(key)).toBe(JSON.stringify(postPayload[key]));
        });
```

So future readers don't have to wonder what's the purpose of this function.

Another option is to use `Object.entries`:

```ts
        Object.entries(postPayload).forEach(([key, value]) => {
          expect(body.get(key)).toBe(JSON.stringify(value));
        });
```

Reply:
Using the function can infer `T`. Otherwise it has to be

```ts
(Object.keys(object) as (keyof typeof object)[]).forEach(...)
```

Not sure if having casting in every occurrence like this is better or having a helper is better. I had to cast the return type of `Object.keys()` in many places already. Now considering adding this function as part of the `core`.

`Object.entries()` has the same issue with `Object.keys()` that `key` is `string`

Reply:
`Object.entries` at least works for this case because `value` would not be `any` but rather the union of all value types in `object`.

Reply:
let me give this a try.

Reply:
converted to `Object.entries`